cmd/zfs/zfs_main.c

```diff
@@ -4383,6 +4383,90 @@ zfs_do_snapshot(int argc, char **argv)
 	return (-1);
 }
 
+typedef struct zfs_send_exclude_arg {
+	size_t count;
+	char **list;
+} zfs_send_exclude_arg_t;
+
+/*
+ * This function creates the zfs_send_exclude_arg_t
+ * object described above; it can be called multiple
+ * times, and the input can be comma-separated.
+ * This is NOT the most efficient data layout; however,
+ * I couldn't think of a non-pathological case where
+ * it should have more than a couple dozen instances
+ * of excludes. If that turns out to be used in
```

Comment:
It might be simpler to just use a linked list? Not a blocker, just a thought. The code should be simpler if we don't have to constantly realloc and track the size and everything.

Reply:
Meh.  Makes the freeing that I just simplified harder ;).  My intent was to sort the array (and do a binary search) at some point if there were complaints about performance, but I most likely was overthinking again.